gen/briefinggen.py

```diff
@@ -24,19 +31,144 @@ class CommInfo:
     freq: RadioFrequency
 
 
+class FrontLineInfo:
+    def __init__(self, front_line: FrontLine):
+        self.front_line: FrontLine = front_line
+        self.player_base: ControlPoint = front_line.control_point_a
+        self.enemy_base: ControlPoint = front_line.control_point_b
+        self.player_zero: bool = self.player_base.base.total_armor == 0
+        self.enemy_zero: bool = self.enemy_base.base.total_armor == 0
+        self.advantage: bool = self.player_base.base.total_armor > self.enemy_base.base.total_armor
+        self.stance: CombatStance = self.player_base.stances[self.enemy_base.id]
+
+    @property
```

Comment:
All of the properties here could be moved to the template.

Reply:
They can be- but you end up with a long chain of object.object.object.object.object.  I broke them out to help readability, can access the attributes directly if preferred.